addons/survey/models/survey_survey.py

```diff
@@ -583,9 +583,32 @@ def _get_pages_or_questions(self, user_input):
             else:
                 result = self.question_and_page_ids.filtered(
                     lambda question: not question.is_page or not is_html_empty(question.description))
+                result = self._get_questions_except_invalid_conditionals(result)
 
         return result
 
+    def _get_questions_except_invalid_conditionals(self, questions):
+        """This method ignores conditional questions that are placed before their trigger"""
```

Comment:
It doesn't really ignore them, right? How about "excludes" or "filters out"?
```suggestion
        """This method excludes conditional questions that are placed before their trigger."""
```